src/community/elasticsearch/src/main/java/org/geoserver/elasticsearch/ElasticConfigurationPage.java

```diff
@@ -305,10 +305,28 @@ protected Component getComponentForProperty(
                                 return new Fragment(id, "empty", ElasticConfigurationPage.this);
                             }
                         } else if (property == ElasticAttributeProvider.DATE_FORMAT) {
-                            if (att.getDateFormat() != null) {
+                            List<String> validFormats = null;
+                            if (att.getValidDateFormats() == null) {
+                                if (att.getDateFormat() != null) {
+                                    validFormats = new ArrayList<String>();
+                                    validFormats.add(att.getDateFormat());
+                                    att.setValidDateFormats(validFormats);
+                                }
+                            } else {
+                                validFormats = att.getValidDateFormats();
+                            }
```

Comment:
On the mailing list I discussed about using a Strategy pattern to interpret the date behaviour. However, due to how small the code sections are that use the date format, I found this way of handling it to be pretty straightforward, but I'm not sure it's the correct approach.

It does work that will be unnecessary for all future DataStores. Essentially this just "corrects" existing DataStores that were using the old date format. It could even be removed in a later release. Otherwise, I am not sure how old DataStores would ever get updated to use the new date format behaviour.

The Strategy option discussed would also work and be a permanent solution, but it wouldn't update old DataStores as far as I can tell. The old DataStores with the old date behaviour is technically incorrect and can fail to interpret Dates correctly. We could leave them alone and instruct users to recreate the DataStores with the new way of handling date formats (which would be handled via the Strategy, allowing both to exist simultaneously).

I am not sure which approach is better here, but hopefully I made the intent clear. If this way is not recommended, and using the Strategy is more proper, I'll update to that.